lib/std/os.zig

```diff
@@ -3111,6 +3111,67 @@ pub fn send(
     return sendto(sockfd, buf, flags, null, 0);
 }
 
+pub const SendFileError = error{
+    /// There was an unspecified error while reading from infd.
+    InputOutput,
+
+    /// There was insufficient memory for reading from infd.
+    NoMem,
+
+    /// The value provided for count overflows the maximum size of either
+    /// infd or outfd.
+    Overflow,
+
+    /// Offset was provided, but infd is not seekable.
+    Unseekable,
+
+    /// The outfd is marked nonblocking and the requested operation would block, and
+    /// there is no global event loop configured.
+    WouldBlock,
+} || UnexpectedError;
+
+/// Transfer data between file descriptors.
+///
+/// The `sendfile` call copies `count` bytes from one file descriptor to another within the kernel. This can
+/// be more performant than transferring data from the kernel to user space and back, such as with
+/// `read` and `write` calls.
+///
+/// The `infd` should be a file descriptor opened for reading, and `outfd` should be a file descriptor
+/// opened for writing. Copying will begin at `offset`, if not null, which will be updated to reflect
+/// the number of bytes read. If `offset` is null, the copying will begin at the current seek position,
+/// and the file position will be updated.
+pub fn sendfile(outfd: fd_t, infd: fd_t, offset: ?*u64, count: usize) SendFileError!usize {
+    while (true) {
+        var rc: usize = undefined;
+        var err: usize = undefined;
+        if (builtin.os == .linux) {
```

Comment:
This should just be an unconditional `system.sendfile`. If someone attempts to call this function on a system without it then they will get a compile error indicating as much.